File: 230-kth-smallest-element-in-a-bst/230-kth-smallest-element-in-a-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
     
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
    
        def inorder(node):
            if not node:
                return 
            inorder(node.left)
            self.k-=1
            if(self.k==0):
                self.ans=node.val
            inorder(node.right)
        
        self.ans=None
        self.k=k
        inorder(root)
        return self.ans
    
    
            
        
        

File: 230-kth-smallest-element-in-a-bst/230-kth-smallest-element-in-a-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
     int ans=0,cnt=0;
    private void lar(TreeNode root,int k){
        
       if(root==null)
         return;
      
       lar(root.left,k);
         cnt++;
         if(k==cnt)
              ans=root.val;
        
       lar(root.right,k);
   }
    public int kthSmallest(TreeNode root, int k) {
       lar(root,k);
       return ans;
    }
}